### PR TITLE
docs for the Grafana Agent Operator

### PR DESCRIPTION
**Pull request by rfratto**
Adds docs for the Grafana Agent Operator introduced in #549.

### CODE REVIEW

File: docs/operator/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      app: grafana-agent-operator
+  template:
+    metadata:
+      labels:
+        app: grafana-agent-operator
+    spec:
+      serviceAccountName: grafana-agent-operator
+      containers:
+      - name: operator
+        image: grafana/agent-operator:v0.15.0
```
Comment (rfratto) [inline docs/operator/getting-started.md:44]: (This image doesn't exist yet, if you're going to test this out locally you'll need to `make agent-operator-image` , re-tag it, and apply it to the k3d cluster using `k3d image import -c agent-operator grafana/agent-operator:v0.15.0`)

File: docs/operator/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Grafana Agent Operator
+
+The Grafana Agent Operator is a Kubernetes operator that makes it easier to
+deploy the Grafana Agent and easier to collect telemetry data from your pods.
+
+Metric collection is based on the [Prometheus
+Operator](https://github.com/prometheus-operator/prometheus-operator) and
+supports the official v1 ServiceMonitor, PodMonitr, and Probe CRDs from the
```
Comment (mapno) [inline docs/operator/README.md:8]:
```suggestion
supports the official v1 ServiceMonitor, PodMonitor, and Probe CRDs from the
```

File: docs/operator/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Grafana Agent Operator
```
Comment (mapno) [inline docs/operator/README.md:1]: I think this is missing a brief description of what the operator actually does, e.g. `Manages kubernetes resources of the Grafana Agent. This is useful for [...].`

File: docs/operator/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
+# Getting Started
+
+An official Helm chart is planned to make it really easy to deploy the Grafana Agent
+Operator on Kubernetes. For now, things must be done a little manually.
+
+## Deploying CustomResourceDefinitions
```
Comment (mapno) [inline docs/operator/getting-started.md:6]: I'm not very familiar with the operator pattern and I was a bit lost with this part at first. I think mentioning that we're creating resources to get the operator running would clarify it.

File: docs/operator/architecture.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+box, and the Grafana Agent Operator defaults support these two systems.
+
+The total number of created metrics pods will be product of `numShards *
+numReplicas`.
+
+## Labels
+
+Two labels are added by default to every metric:
+
+- `cluster`, representing the `GrafanaAgent` deployment. Holds the value of
```
Comment (mapno) [inline docs/operator/architecture.md:91]: Does this override the cluster label from k8s SD?

**Review by mapno — COMMENTED** [review]
LGTM overall, just left some minor comments

File: docs/operator/architecture.md (hunk 5 above)
Comment (rfratto) [reply docs/operator/architecture.md:91]: No, k8s SD doesn't actually have a cluster label, but it's a common pattern to manually add one

**Pull request merged**